File: poultry-pal-service/src/main/java/co/za/hlaluko/dynamics/poultry/pal/model/dto/GrowingPhase.java
package co.za.hlaluko.dynamics.poultry.pal.model.dto;

import lombok.Getter;

@Getter
public enum GrowingPhase {
  BROODING_PHASE("Brooding Phase"),
  GROWING_REARING_PHASE("Growing Phase"),
  PRODUCTION_FINISHING_PHASE("Production or Finishing Phase");

  private final String value;

  GrowingPhase(String value) {
    this.value = value;
  }

    public static GrowingPhase fromValue(String value) {
    for (GrowingPhase coopType : GrowingPhase.values()) {
      if (coopType.value.equalsIgnoreCase(value) || coopType.name().equalsIgnoreCase(value)) {
        return coopType;
      }
    }
    throw new IllegalArgumentException("Invalid growing phase value: " + value);
  }
}

File: poultry_pal_plus_app/lib/models/growing_phase.dart
enum GrowingPhase {
  NONE("Select growth phase"),
  BROODING_PHASE("Brooding Phase"),
  GROWING_REARING_PHASE("Growing Phase"),
  PRODUCTION_FINISHING_PHASE("Production or Finishing Phase");

  final String value;

  const GrowingPhase(this.value);

  static GrowingPhase fromValue(String value) {
    for (var phase in GrowingPhase.values) {
      if (phase.value == value || phase.name.toString() == value) {
        return phase;
      }
    }
    throw ArgumentError("Invalid phase value {$value}");
  }
}